### PR TITLE
{tools}[foss/2015a,GCC/4.9.2] numactl 2.0.10 (OK)

### PR DESCRIPTION
**Pull request by rjeschmi**
This should work to get foss-2015a back working.

### CODE REVIEW

**Comment by rjeschmi** [comment]
Test report by @rjeschmi
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux centos 6.5, AMD Opteron(tm) Processor 6380, Python 2.7.8
See https://gist.github.com/e565ce40b623fc95387c for a full test report.

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/n/numactl/numactl-2.0.10-GCC-4.9.2.eb [hunk 1]
```diff
@@ -11,6 +13,10 @@ The libnuma library provides convenient ways for you to add NUMA memory policies
 sources = [SOURCE_TAR_GZ]
 source_urls = ['ftp://oss.sgi.com/www/projects/libnuma/download/']
 
+dependencies = [('Automake', '1.15'),('libtool', '2.4.2')]
```
Comment (boegel) [inline easybuild/easyconfigs/n/numactl/numactl-2.0.10-GCC-4.9.2.eb]: put these on separate lines please? why is the libtool required exactly, is it used by `autogen`?
also, these are only build dependencies?

```python
builddependencies = [
    ('Automake', '1.15'),
    ('libtool', '2.4.2'),
]
```

File: easybuild/easyconfigs/l/libtool/libtool-2.4.2-GCC-4.9.2.eb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+homepage = 'http://www.gnu.org/software/libtool'
+description = """GNU libtool is a generic library support script. Libtool hides the complexity of using shared libraries
+behind a consistent, portable interface."""
+toolchain = {'name': 'GCC', 'version': '4.9.2'} # tweaked by EasyBuild (was: {'name': 'GCC', 'version': '4.8.2'})
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = [GNU_SOURCE]
+
+moduleclass = 'lib'
+
```
Comment (boegel) [inline easybuild/easyconfigs/l/libtool/libtool-2.4.2-GCC-4.9.2.eb]: drop empty line at the end

File: easybuild/easyconfigs/l/libtool/libtool-2.4.2-GCC-4.9.2.eb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Built with EasyBuild version 2.0.0dev on 2015-01-21_18-09-07
+easyblock = 'ConfigureMake'
+
+name = 'libtool'
+version = '2.4.2'
+
+homepage = 'http://www.gnu.org/software/libtool'
+description = """GNU libtool is a generic library support script. Libtool hides the complexity of using shared libraries
+behind a consistent, portable interface."""
+toolchain = {'name': 'GCC', 'version': '4.9.2'} # tweaked by EasyBuild (was: {'name': 'GCC', 'version': '4.8.2'})
```
Comment (boegel) [inline easybuild/easyconfigs/l/libtool/libtool-2.4.2-GCC-4.9.2.eb]: remove the comment, add empty line above `toolchain` line

File: easybuild/easyconfigs/l/libtool/libtool-2.4.2-GCC-4.9.2.eb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Built with EasyBuild version 2.0.0dev on 2015-01-21_18-09-07
```
Comment (boegel) [inline easybuild/easyconfigs/l/libtool/libtool-2.4.2-GCC-4.9.2.eb]: drop this comment please

File: easybuild/easyconfigs/n/numactl/numactl-2.0.10-GCC-4.9.2.eb (hunk 1 above)
Comment (rjeschmi) [inline easybuild/easyconfigs/n/numactl/numactl-2.0.10-GCC-4.9.2.eb]: without the libtool dep I was getting an error:

```
== 2015-01-21 18:11:56,158 main.run ERROR EasyBuild crashed with an error (at easybuild/tools/run.py:382 in parse_cmd_output): cmd "./autogen.sh &&  ./configure --prefix=/scratch/ro
b/eb-dev-root/easybuild/software/numactl/2.0.10-GCC-4.9.2 " exited with exitcode 1 and output:
configure.ac:15: installing 'build-aux/compile'
configure.ac:9: installing 'build-aux/install-sh'
configure.ac:9: installing 'build-aux/missing'
Makefile.am:8: error: Libtool library used but 'LIBTOOL' is undefined
Makefile.am:8:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
Makefile.am:8:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
Makefile.am:8:   If 'LT_INIT' is in 'configure.ac', make sure
Makefile.am:8:   its definition is in aclocal's search path.
Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
autoreconf: automake failed with exit status: 1
```

Comment (boegel) [inline easybuild/easyconfigs/n/numactl/numactl-2.0.10-GCC-4.9.2.eb]: ok, so both should be build deps indeed

**Comment by rjeschmi** [comment]
It is sort of annoying that the comments are hidden after I pushed the update.

anyway, to repeat, libtool seems to be required or autogen.sh dies

File: easybuild/easyconfigs/n/numactl/numactl-2.0.10-GCC-4.9.2.eb [hunk 5]
```diff
@@ -11,6 +13,13 @@ The libnuma library provides convenient ways for you to add NUMA memory policies
...
 source_urls = ['ftp://oss.sgi.com/www/projects/libnuma/download/']
 
+builddependencies = [
+    ('Automake', '1.15'),
+    ('libtool', '2.4.2'),
+]
+
+preconfigopts = './autogen.sh && '
+
 toolchain = {'name': 'GCC', 'version': '4.9.2'}
```
Comment (boegel) [inline easybuild/easyconfigs/n/numactl/numactl-2.0.10-GCC-4.9.2.eb]: nitpicking: we usually place the `toolchain` line above the `sources`/`source_urls` lines, with an empty line in between... (not sure why it's below the `sources` line originally, it shouldn't be)

File: easybuild/easyconfigs/n/numactl/numactl-2.0.10-GCC-4.9.2.eb [hunk 6]
```diff
@@ -3,6 +3,8 @@ easyblock = 'ConfigureMake'
 name = 'numactl'
 version = '2.0.10'
 
+checksums = ['682c38305b2596967881f3d77bc3fc9c']
+
 homepage = 'http://oss.sgi.com/projects/libnuma/'
 description = """The numactl program allows you to run your application program on specific cpu's and memory nodes.
 It does this by supplying a NUMA memory policy to the operating system before running your program.
```
Comment (boegel) [inline easybuild/easyconfigs/n/numactl/numactl-2.0.10-GCC-4.9.2.eb]: while you're at it: leading space here, and the same for any other 'continuation' lines of the `description`

File: easybuild/easyconfigs/l/libtool/libtool-2.4.2-GCC-4.9.2.eb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+easyblock = 'ConfigureMake'
+
+name = 'libtool'
+version = '2.4.2'
+
+homepage = 'http://www.gnu.org/software/libtool'
+description = """GNU libtool is a generic library support script. Libtool hides the complexity of using shared libraries
+behind a consistent, portable interface."""
```
Comment (boegel) [inline easybuild/easyconfigs/l/libtool/libtool-2.4.2-GCC-4.9.2.eb]: leading space pls

**Comment by boegel** [comment]
@rjeschmi: you can still access the comments via the "outdated diff" links, but yes, something it's annoying (but it's what you want when iterating over the PR, and fixing remarks...)

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/abaec36c1a23b2b47c16 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e1a2a09e87d673aac6bb for a full test report.

**Comment by boegel** [comment]
@rjeschmi: test failed due to old source tarball on my end with a different MD5, trying again now with that tarball removed/renamed..

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/cbbfd1e51a78feb454ba for a full test report.

**Comment by rjeschmi** [comment]
At least checksum validation passed :)

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/1ca810eeec82ffaf2ff8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/923a5cd30d75d6244710 for a full test report.

**Comment by boegel** [comment]
successful test report after I remember how to remove a file (urgh)...

Looks good to go now, thanks for looking into this @rjeschmi!

**Comment by boegel** [comment]
For historical purposes: this PR is requires because the source tarball for numactl v2.0.10 changed somewhere in the last couple of weeks. My tarball which was downloaded Dec 9th 2014, has a different MD5sum (`58443c6fe7e67126de07bd86e4d6b825`), and included a readily available `configure` script, while the current tarball available upstream doesn't, and requires running the `autogen.sh` script as is done now.

Differences between the two versions:

```
$ diff -ru numactl-2.0.10.20141209 numactl-2.0.10
Only in numactl-2.0.10.20141209: Makefile.in
Only in numactl-2.0.10.20141209: aclocal.m4
Only in numactl-2.0.10: autogen.sh
Only in numactl-2.0.10.20141209: build-aux
Only in numactl-2.0.10.20141209: config.h.in
Only in numactl-2.0.10.20141209: configure
Only in numactl-2.0.10/m4: .gitignore
Only in numactl-2.0.10.20141209/m4: libtool.m4
Only in numactl-2.0.10.20141209/m4: ltoptions.m4
Only in numactl-2.0.10.20141209/m4: ltsugar.m4
Only in numactl-2.0.10.20141209/m4: ltversion.m4
Only in numactl-2.0.10.20141209/m4: lt~obsolete.m4
Only in numactl-2.0.10: manlinks
Only in numactl-2.0.10: mkolddemo
Only in numactl-2.0.10: numamon.c
Only in numactl-2.0.10: stream_main.c
Only in numactl-2.0.10/test: getnodemask.c
Only in numactl-2.0.10/test: tshm.c
Only in numactl-2.0.10: threadtest.c
```

cc: @hajgato, @sbecuwe, @hpcleuven, @gjbex (since this concerns the common toolchain `foss/2015a`)

**Pull request merged**